notebooks/hubble_resid.ipynb

```diff
@@ -0,0 +1,166 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Plotting Hubble Residuals\n",
+    " \n",
+    "Creates Hubble diagram and plots residuals"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import scipy\n",
+    "from astropy.cosmology import FlatLambdaCDM\n",
+    "from astropy import units as u\n",
+    "from scipy.stats import chisquare\n",
+    "from scipy.optimize import minimize\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sdss_combined = pd.read_csv('../results/sdss_combined.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finding a Best Fit Cosmology"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Om0_assumed = 0.315\n",
+    "redshift = sdss_combined['zCMB']\n",
+    "dist_mod = sdss_combined['muSALT2zspec'].to_numpy()\n",
+    "dist_mod_err = sdss_combined['muerrSALT2zspec']\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@np.vectorize\n",
+    "def dist_mod_chisq(H0):\n",
+    "    cosmo = FlatLambdaCDM(H0=H0 * u.km / u.s / u.Mpc, Om0=Om0_assumed)\n",
+    "    model_dist_mod = cosmo.distmod(redshift)\n",
+    "    chisq, p = chisquare(dist_mod * u.mag, model_dist_mod)\n",
```

Comment:
This chi-squared function is not the same as the one we want to use. I realize the different definitions of chi-squared can be confusing.

Here is a reference: http://www.physics.csbsju.edu/stats/chi-square.html

The function currently being used is at the bottom of the linked page. We want to use the equation at the top (use the error values for the distance moduli in the demoninator)